src/field_player_avatar.c

```diff
@@ -1828,30 +1867,21 @@ static bool8 Fishing_CheckForBite(struct Task *task)
     if (!DoesCurrentMapHaveFishingMons())
     {
         task->tStep = FISHING_NO_BITE;
+        return TRUE;
     }
-    else
-    {
-        if (!GetMonData(&gPlayerParty[0], MON_DATA_SANITY_IS_EGG))
-        {
-            u16 ability = GetMonAbility(&gPlayerParty[0]);
-            if (ability == ABILITY_SUCTION_CUPS || ability  == ABILITY_STICKY_HOLD)
-            {
-                if (Random() % 100 > 14)
-                    bite = TRUE;
-            }
-        }
 
-        if (!bite)
-        {
-            if (Random() & 1)
-                task->tStep = FISHING_NO_BITE;
-            else
-                bite = TRUE;
-        }
+    if(Fishing_DoesFirstMonInPartyHaveSuctionCupsOrStickyHold())
+        bite = Fishing_CheckForBiteWithStickyHold();
```

Comment:
This applies to the whole but why not call `Fishing_RollForBite(TRUE);` directly?

Reply:
`Fishing_CheckForBiteWithStickyHold` tells everybody exactly what is _supposed_ to happen on that line, making it easier to understand / debug / refactor. Sure, you could make some assumptions with `Fishing_RollForBite(TRUE)`, but why make anybody assume when you could just be clear?

Especially in codebase where the primary user is somebody who has never coded in their lives, I think expansion should _try_ to be easier to read than not.

Reply:
https://github.com/rh-hideout/pokeemerald-expansion/pull/4343/commits/3a679cc66fd33e8000297c8646d58cc09360cf3e